editor/components/inserter/index.js

```diff
@@ -37,7 +37,7 @@ class Inserter extends wp.element.Component {
 					label={ wp.i18n.__( 'Insert block' ) }
 					onClick={ this.toggle }
 					className="editor-inserter__toggle" />
-				{ opened && <InserterMenu position={ position } onSelect={ this.close } /> }
+				{ opened && <InserterMenu position={ position } onSelect={ this.close } onClickOutside={ this.close } /> }
```

Comment:
Could we merge the `onSelect` and the `onClickOutside` prop into a single prop `onClose` that takes a nullable block parameter?